lib/runner.js

```diff
@@ -443,11 +443,22 @@ Runner.prototype.fail = function (test, err, force) {
     err = thrown2Error(err);
   }
 
-  try {
-    err.stack =
-      this.fullStackTrace || !err.stack ? err.stack : stackFilter(err.stack);
-  } catch (ignore) {
-    // some environments do not take kindly to monkeying with the stack
+  // Filter the stack traces
+  if (!this.fullStackTrace) {
+    const alreadyFiltered = new Set();
+    let currentErr = err;
+
+    while (currentErr?.stack && !alreadyFiltered.has(currentErr)) {
```

Comment:
> `?.`

From https://github.com/mochajs/mocha/actions/runs/8068123423/job/22040196920?pr=4829:

```plaintext
nps is executing `build` : rollup -c ./rollup.config.js

./browser-entry.js → ./mocha.js...
[!] (plugin at position 4) SyntaxError: Unexpected token (467:22) in /home/runner/work/mocha/mocha/lib/runner.js
lib/runner.js (467:22)
SyntaxError: Unexpected token (467:22) in /home/runner/work/mocha/mocha/lib/runner.js
    at Parser.pp$4.raise (/home/runner/work/mocha/mocha/node_modules/rollup-plugin-node-globals/node_modules/acorn/dist/acorn.js:2757:13)
```

@voxpelli what do you want to do here? My intuition is to delay using `?.` until we redo the rollup builds / etc..

Reply:
I agree, removing it

Reply:
I am now 0 for 2 in making suggestions on this PR. Not great for me. 😂